common/web/gesture-recognizer/src/engine/configuration/gestureRecognizerConfiguration.ts

```diff
@@ -81,4 +84,34 @@ export interface GestureRecognizerConfiguration<HoveredItemType> {
    * @returns
    */
   readonly itemIdentifier?: (coord: Omit<InputSample<any>, 'item'>, target: EventTarget) => HoveredItemType;
+}
+
+export function preprocessRecognizerConfig<HoveredItemType>(
+  config: GestureRecognizerConfiguration<HoveredItemType>
+): Nonoptional<GestureRecognizerConfiguration<HoveredItemType>> {
+  // Allows configuration pre-processing during this method.
+  let processingConfig: Mutable<Nonoptional<GestureRecognizerConfiguration<HoveredItemType>>> = {...config} as Nonoptional<GestureRecognizerConfiguration<HoveredItemType>>;
+  processingConfig.mouseEventRoot = processingConfig.mouseEventRoot ?? processingConfig.targetRoot;
+  processingConfig.touchEventRoot = processingConfig.touchEventRoot ?? processingConfig.targetRoot;
+
+  processingConfig.inputStartBounds = processingConfig.inputStartBounds ?? processingConfig.targetRoot;
+  processingConfig.maxRoamingBounds = processingConfig.maxRoamingBounds ?? processingConfig.targetRoot;
+  processingConfig.safeBounds       = processingConfig.safeBounds       ?? new PaddedZoneSource([2]);
+
+  processingConfig.itemIdentifier   = processingConfig.itemIdentifier   ?? (() => null);
+
+  if(!config.paddedSafeBounds) {
+    let paddingArray = config.safeBoundPadding;
+    if(typeof paddingArray == 'number') {
+      paddingArray = [ paddingArray ];
+    }
+    paddingArray = paddingArray ?? [3];
```

Comment:
Why 3? Magic?

Reply:
https://github.com/keymanapp/keyman/blob/be196744fd1fef9baf138072b2a5e44c95da6d7b/common/web/gesture-recognizer/src/engine/configuration/gestureRecognizerConfiguration.ts#L58-L59

From:

https://github.com/keymanapp/keyman/blob/be196744fd1fef9baf138072b2a5e44c95da6d7b/common/web/gesture-recognizer/src/engine/configuration/gestureRecognizerConfiguration.ts#L52-L63

If memory serves, this default is derived from the corresponding value KMW is already using for the OSK.  I've just abstracted its purpose and given it a name and documentation.